src/common/utils.cpp

```diff
@@ -818,6 +818,43 @@ std::string trim(std::string const& str, std::string const& whitespace)
     return str.substr(strBegin, strRange);
 }
 
+/// Matches the given target with the given pattern, using the algorithm linked below.
+/// There's possibly a more clever / modern c way of doing this.
+/// https://www.geeksforgeeks.org/wildcard-character-matching/
+bool matchesRecursive(const char* pattern, const char* target, const char* wildcard) {
```

Comment:
While I don't really like this and would have preferred regex, I get that most people tend to be happier using wildcards instead of busting out regex for every little thing. It's also unfortunate that Lua's pattern matching is a stripped-down version of regex (as far as I know) and isn't really suitable.